### DISCUSSION TITLE
onCreate method

### DISCUSSION THREAD

**Discussion by picobulgarelli** [skydoves/Balloon#175] [Q&A] [answered]
Hello I noticed there's no onCreate or onShow method, is correct?

I need to call a function when one of those method are called.

**Comment by skydoves** [comment, marked as answer]
Could you explain more details?
I think we can just call on the `onCreate` method.

**Reply by picobulgarelli** [reply]
I instantiate the class like this:

`private val dialog by balloon(Dialog::class)`

and then I can access its method like setOnBalloonDismissListener in this way:

```
dialog.setOnBalloonDismissListener {
 // do something
}
```

but onCreate method is not available.

**Reply by skydoves** [reply]
Oh, I see!
Then we can use this method.

```kotlin
dialog.setOnBalloonInitializedListener {
 // do something after initialized the balloon.
}
```

**Reply by picobulgarelli** [reply]
Thank you for your quick reply but that method doesn't exist. I can only access to the var _onBalloonInitializedListener_ and it could be ok if I set = this and I implement the method in my activity.

![Schermata 2021-03-25 alle 12 07 07](https://user-images.githubusercontent.com/48985589/112463562-b224ce00-8d62-11eb-8078-40c14f0c9932.png)

**Reply by skydoves** [reply]
Ah, I just know that method only exists on Builder.
You can implement it this way instead.

```kotlin
balloon.onBalloonInitializedListener = OnBalloonInitializedListener {
  // do something
}
```

I will add new methods related to `OnBalloonInitializedListener` in the next release.
Thanks!

**Reply by picobulgarelli** [reply]
Perfect, thank you
